### PR TITLE
Add `--repository-slug` flag to CLI, and `repositorySlug` + `branchName` options to Action

### PR DESCRIPTION
**Pull request by ghengeveld**
This allows customers to override the repository owner and/or name through either the `repositorySlug` flag/option, or the `branchName` flag/option, whichever is more convenient. Either of them can affect the `ownerName`, which is used to determine whether a build originates from a fork.

We didn't previously expose the `branchName` option on the GitHub Action, now we do.

For example:
- `repositorySlug: "chromaui/chromatic"` sets the owner name to `chromaui` and repository name to `chromatic`
- `branchName: "chromaui:feature"` sets the owner name to `chromaui` and the branch name to `feature`.

If both specify an owner name, they must match.

### CODE REVIEW

File: bin-src/lib/getOptions.ts [hunk 1]
```diff
@@ -32,7 +27,8 @@ const undefinedIfEmpty = <T>(array: T[]) => {
 export default function getOptions({ argv, env, flags, log, packageJson }: Context): Options {
   const fromCI = !!flags.ci || !!process.env.CI;
   const [patchHeadRef, patchBaseRef] = (flags.patchBuild || '').split('...').filter(Boolean);
-  const [branchName, ownerName] = (flags.branchName || '').split(':').reverse();
+  const [branchName, branchOwner] = (flags.branchName || '').split(':').reverse();
+  const [repositoryOwner] = flags.repositorySlug ? flags.repositorySlug.split('/') : [];
```
Comment (JonathanKolnik) [inline bin-src/lib/getOptions.ts:31]: Can we write a test for this?

Comment (JonathanKolnik) [reply bin-src/lib/getOptions.ts:31]: and maybe how it relates to which one takes precedence on line 67

**Review by thafryer — APPROVED** [review]
This looks good. I noticed we have a few more flags missing in the GitHub Action, like the JUnit report one. We should formalize that anytime we add an option to the CLI we add it to the Action unless circumstances say we shouldn't.

File: bin-src/lib/getOptions.ts (hunk 1 above)
Comment (ghengeveld) [reply bin-src/lib/getOptions.ts:31]: Done

File: action-src/main.ts [hunk 2]
```diff
@@ -136,68 +136,73 @@ async function run() {
 
   try {
     // Remember to keep this list in sync with ../action.yml
```
Comment (itwasmattgregg) [inline action-src/main.ts:138]: 👍

File: action.yml [hunk 3]
```diff
@@ -6,35 +6,44 @@ branding:
   color: 'orange'
 
 inputs:
```
Comment (itwasmattgregg) [inline action.yml:8]: Would be sweet if this auto-generated the docs page.

**Review by itwasmattgregg — APPROVED** [review]
This all looks good. Curious though what happens if you specify `repositorySlug` with a different owner than what gets generated from `branchName` as the owner.

File: action.yml (hunk 3 above)
Comment (ghengeveld) [reply action.yml:8]: Our docs are based on Markdown, and live in a different repository. It's definitely possible, but not really worth the effort and complexity. We don't change options all that often.

**Comment by ghengeveld** [comment]
> This all looks good. Curious though what happens if you specify `repositorySlug` with a different owner than what gets generated from `branchName` as the owner.

There's a check to verify that if both are specified (and `branchName` contains an owner name prefix), they must match. Otherwise an error is thrown and no build will run.

**Review by JonathanKolnik — APPROVED** [review]

**Pull request merged**